src/TestFramework/XML/SafeDOMXPath.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\TestFramework\XML;
+
+use DOMDocument;
+use DOMElement;
+use DOMNameSpaceNode;
+use DOMNode;
+use DOMNodeList;
+use DOMXPath;
+use Error;
+use InvalidArgumentException;
+use function sprintf;
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal
+ *
+ * @property DOMDocument $document
+ */
+final readonly class SafeDOMXPath
+{
+    private DOMXPath $xPath;
+
+    public function __construct(
+        public DOMDocument $document,
+    ) {
+        $this->xPath = new DOMXPath($document);
+    }
+
+    public static function fromFile(string $pathname): self
```

Comment:
extractable change maybe? The old approach does a file_get_contents + loadXML, not sure which one is better

Reply:
- [x] Evaluate the impact: https://github.com/infection/infection/pull/2561
- [x] Introduce `::fromFile()`: https://github.com/infection/infection/pull/2558
- [ ] Leverage `::fromFile()`
- [ ] Extract the methods instead of `__get`